internal/statemachine/classic_states.go

```diff
@@ -118,39 +118,80 @@ func (stateMachine *StateMachine) runLiveBuild() error {
 }
 
 // prepareClassicImage calls image.Prepare to seed extra snaps in classic images
-/*func (stateMachine *StateMachine) prepareClassicImage() error {
+// currently only used when --filesystem is provided
+func (stateMachine *StateMachine) prepareClassicImage() error {
 	var classicStateMachine *ClassicStateMachine
 	classicStateMachine = stateMachine.parent.(*ClassicStateMachine)
 
-	var imageOpts image.Options
+	if classicStateMachine.Opts.Filesystem != "" {
+		var imageOpts image.Options
 
-	var err error
-	imageOpts.Snaps, imageOpts.SnapChannels, err = parseSnapsAndChannels(
-		classicStateMachine.commonFlags.Snaps)
-	if err != nil {
-		return err
-	}
+		var err error
+		imageOpts.Snaps, imageOpts.SnapChannels, err = parseSnapsAndChannels(
```

Comment:
I'm actually wondering now - should we maybe skip all this if the list returned from `parseSnapsAndChannels()` is empty? Since from what I see, right now even if there are no snaps added on the commandline, ubuntu-image still clears out the preseed and adds it afresh.
...or maybe that's needed to get the customizations going? Though that feels more like an Ubuntu Core feature.

Reply:
You're right that this seems like an unnecessary extra step that could end up causing problems. I have a question about expected behavior for an edge case though:

Let's say a user is making an ubuntu image using `--filesystem` on an already pre-seeded filesystem. They don't want to install any additional snaps, so they don't use the `--snap` flag. However, they pass the `--channel` flag and give it an argument like `--channel=latest/edge`. In this scenario, should we remove the pre-seeded snaps and re-seed them from the channel that is specified on the command line?